CONTRIBUTING.rst

```diff
@@ -0,0 +1,480 @@
+
+The DCO is an attestation attached to every contribution made by every
+developer. In the commit message of the contribution, (described more fully
+later in this document), the developer simply adds a ``Signed-off-by``
+statement and thereby agrees to the DCO.
+
+When a developer submits a patch, is a committment that the contributor has
+the right to submit the patch per the license.  The DCO agreement is shown
+below and at http://developercertificate.org/.
+
+
+.. code-block:: none
+
+    Developer's Certificate of Origin 1.1
+
+    By making a contribution to this project, I certify that:
+
+    (a) The contribution was created in whole or in part by me and I
+        have the right to submit it under the open source license
+        indicated in the file; or
+
+    (b) The contribution is based upon previous work that, to the
+        best of my knowledge, is covered under an appropriate open
+        source license and I have the right under that license to
+        submit that work with modifications, whether created in whole
+        or in part by me, under the same open source license (unless
+        I am permitted to submit under a different license), as
+        Indicated in the file; or
+
+    (c) The contribution was provided directly to me by some other
+        person who certified (a), (b) or (c) and I have not modified
+        it.
+
+    (d) I understand and agree that this project and the contribution
+        are public and that a record of the contribution (including
+        all personal information I submit with it, including my
+        sign-off) is maintained indefinitely and may be redistributed
+        consistent with this project or the open source license(s)
+        involved.
+
+DCO Sign-Off Methods
+====================
+
+The DCO requires a sign-off message in the following format appear on each
+commit in the pull request::
+
+   Signed-off-by: Zephyrus Zephyr <zephyrus@zephyrproject.org>
+
+The DCO text can either be manually added to your commit body, or you can add
+either ``-s`` or ``--signoff`` to your usual Git commit commands. If you forget
+to add the sign-off you can also amend a previous commit with the sign-off by
+running ``git commit --amend -s``. If you've pushed your changes to GitHub
+already you'll need to force push your branch after this with ``git push -f``.
+
+
+Prerequisites
+*************
+
+.. _Zephyr Project website: https://zephyrproject.org
+
+As a contributor, you'll want to be familiar with the Zephyr project, how to
+configure, install, and use it as explained in the `Zephyr Project website`_
+and how to set up your development environment as introduced in the Zephyr
+`Getting Started Guide`_.
+
+.. _Getting Started Guide:
+   https://www.zephyrproject.org/doc/getting_started/getting_started.html
+
+Linux is the preferred host environment for Zephyr development. As a Linux
+developer, you should be familiar with common tools such as Git and Make, and
+platforms such as GitHub.
```

Comment:
I would remove this, lets not make preferences, our goal is to work well on all OSes